### PR TITLE
Multiple player movement

### PR DESCRIPTION
**Pull request by TaliKert**
There are inconsistencies in the code, especially clientside, for example: 
`Observable<int[]> userLocationObservable = GameService.getInstance().getCharacterLocation();
        Observable<AbstractMessage> opponentLocationObservable = GameService.getInstance().getOpponentLocation();`

I didn't want to go overboard with the pull request so i left those irregularities in for now.

nevertheless, the code is compilable, just run the server and run 2 clients

### CODE REVIEW

File: server/src/main/java/models/MatchModel.java [hunk 1]
```diff
@@ -31,6 +32,33 @@ public MatchModel(Map<Auth.LoginData, QueryHandler> clients) {
...
+    private void updateClientDelayThread() {
+        new Thread(() -> {
+            while (true) {
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+                if (players.size() > 1) {
+                    for (int i = 0; i < players.size(); i++) {
```
Comment (agnessaik) [inline server/src/main/java/models/MatchModel.java]: what exactly is going on here and why?

File: client/src/main/java/scenes/BattleFieldScene.java [hunk 2]
```diff
@@ -80,6 +81,13 @@ public void handle(KeyEvent event) {
 
     private void createObserver() {
         Observable<int[]> userLocationObservable = GameService.getInstance().getCharacterLocation();
+        Observable<AbstractMessage> opponentLocationObservable = GameService.getInstance().getOpponentLocation();
+
+        opponentLocationObservable.subscribe(data -> {
+            Platform.runLater(() -> {
+                showOpponent((Location.UserLocation) data);
+            });
+        });
```
Comment (agnessaik) [inline client/src/main/java/scenes/BattleFieldScene.java]: I would create new method gor oponent observer but you don´t have to chenge it, I think it´s more of a taste issue.

File: server/src/main/java/models/MatchModel.java (hunk 1 above)
Comment (TaliKert) [reply server/src/main/java/models/MatchModel.java]: this checks if there are more than one player in the server, I dont think its neccesary anymore

Comment (agnessaik) [reply server/src/main/java/models/MatchModel.java]: ups I added that in the wrong spot, I actually ment verything going on in that method especially the list action

Comment (agnessaik) [reply server/src/main/java/models/MatchModel.java]: Oh okay, so playersWithoutThemselves stands for opponents? I don´t think that´s the best name, suggest changing it to oponents or otherPlayers or sth like that. 
But are you sure the best solution is creating new list every time, why not just without creating new list sending other players locations and player location in new for loop using if statment? Maybe I miss something important.

**Pull request merged**